vlc-android/src/org/videolan/vlc/gui/network/MRLPanelFragment.kt

```diff
@@ -0,0 +1,135 @@
+
+const val TAG = "VLC/MrlPanelFragment"
+
+class MRLPanelFragment : DialogFragment(), View.OnKeyListener, TextView.OnEditorActionListener, View.OnClickListener, MRLAdapter.MediaPlayerController {
+    private lateinit var adapter: MRLAdapter
+    private lateinit var editText: TextInputLayout
+
+    val isEmpty: Boolean
+        get() = adapter.isEmpty
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setStyle(DialogFragment.STYLE_NO_FRAME, 0)
+    }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        val v = inflater.inflate(R.layout.mrl_panel, container, false)
+        editText = v.findViewById(R.id.mrl_edit)
+        editText.editText?.setOnKeyListener(this)
+        editText.editText?.setOnEditorActionListener(this)
+        val recyclerView = v.findViewById<RecyclerView>(R.id.mrl_list)
+        recyclerView.addItemDecoration(DividerItemDecoration(activity, DividerItemDecoration.VERTICAL))
+        recyclerView.layoutManager = LinearLayoutManager(activity)
+        adapter = MRLAdapter(this)
+        recyclerView.adapter = adapter
+        v.findViewById<View>(R.id.send).setOnClickListener(this)
+        return v
+    }
+
+    override fun onStart() {
+        super.onStart()
+        updateHistory()
+    }
+
+    override fun onSaveInstanceState(outState: Bundle) {
+        super.onSaveInstanceState(outState)
+        outState.putString(KEY_MRL, editText.editText?.text.toString())
+    }
+
+    override fun onViewStateRestored(savedInstanceState: Bundle?) {
+        super.onViewStateRestored(savedInstanceState)
+        val mrl = savedInstanceState?.getString(KEY_MRL)
+        mrl?.let {
+            editText.editText?.setText(mrl)
+        }
+    }
+
+    private fun updateHistory() {
+        adapter.setList(VLCApplication.getMLInstance().lastStreamsPlayed())
+    }
+
+    override fun onKey(v: View, keyCode: Int, event: KeyEvent): Boolean {
+        return (keyCode == EditorInfo.IME_ACTION_DONE ||
+                keyCode == EditorInfo.IME_ACTION_GO ||
+                event.action == KeyEvent.ACTION_DOWN && event.keyCode == KeyEvent.KEYCODE_ENTER) && processUri()
+    }
+
+    private fun processUri(): Boolean {
+        if (!TextUtils.isEmpty(editText.editText?.text)) {
+            val mw = MediaWrapper(Uri.parse(editText.editText?.text.toString().trim()))
+            playMedia(mw)
+            editText.editText?.text?.clear()
+            return true
+        }
+        return false
+    }
+
+    override fun playMedia(mw: MediaWrapper) {
+        mw.type = MediaWrapper.TYPE_STREAM
+        MediaUtils.openMedia(activity, mw)
+        updateHistory()
+        activity?.supportInvalidateOptionsMenu()
+        UiTools.setKeyboardVisibility(editText, false)
+        dismiss()
+    }
+
+    override fun onEditorAction(v: TextView, actionId: Int, event: KeyEvent): Boolean {
```

Comment:
can be shorten too:
`override fun onEditorAction(v: TextView, actionId: Int, event: KeyEvent) = false`